src/timeout/timeout-collection.model.ts

```diff
@@ -60,6 +53,14 @@ export class TimeoutCollection {
 		this._timeoutCollection = [];
 	}
 
+	private _getWrappedHandler(handler: Function): Function {
```

Comment:
This can potentially cause problems when calling mutliple timeouts with the same handler, if a timeout with smaller index is executed after a timeout with a larger index..

Reply:
Good catch.

Unfortunately, I struggled all morning trying to figure out how to wire the "completing" of the `Timeout` objects into the delegation to the default `setTimeout`, and still don't really have any good ideas for how to mitigate this issue... the `id` is the unique field, but I don't have it until after I schedule a timeout, at which point I've already committed to a handler...

Do you have any ideas for how to better lookup the `Timeout` object to remove when the timeout completes?

Reply:
My initial thought on this was to have a `_scheduledTimeoutCollection` in addition to the normal one, and once `timeoutCollection.add()` is called, it would push the `new Timeout` object to both collections, and in addition create an `originalSetTimeout` (same as window's one) to remove that timeout from `_scheduledTimeoutCollection` with the same time delay.

I do like your solution better though, if we can find another way to distinguish the timeouts without having the `id` (it's late and I have no idea as for now haha)

Reply:
I've had a brain blast of sorts...

We _could_ maintain a time-events-manager-unique UUID in each `Timeout`. We could generate the UUID as part of the Timeout constructor, and use that UUID to reference the `Timeout` from the "completion" logic. Then, we could set the normal `id` on the `Timeout` whenever we get it back from the default `setTimeout`, and use that id when we delegate to the default `clearTimeout` method.

What do you think?

If you're ameniable to the idea, how do you feel about requiring other dependencies, such as https://github.com/kelektiv/node-uuid ? I _could_ add a psuedo-UUID generator, but I'd rather rely on one that's pretty robust to get truly unique UUIDs...

Reply:
I tried avoiding that at first since the original objects already have a UUID, but seems like it's the only reasonable way if we want to implement the status solution.

@bargoldi anything to add?